src/backend/api/reservations.js

```diff
@@ -0,0 +1,97 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+// GET Returns all reservations
+router.get("/", async (req, res) => {
+  try {
+    const data = await knex("Reservation").select();
+    res.status(200).json(await data);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: "Internal Server Error" });
+  }
+});
+
+// POST Adds a new reservation to the database
+router.post("/", async (req, res) => {
+  try {
+    const newReservation = req.body;
+    const [id] = await knex("Reservation").insert(newReservation);
```

Comment:
This is a [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) of the returned array, This means that you have the actual value of the id in a const called `id`.
Therefore, for the check on the next line you can simply put in `if(id)`

Reply:
It's hard to remember why i did that, but i remember that reason was that knex.raw returns array in array. Probably i deleted the part of code that was connected with it somehow 🤷‍♂️🙆‍♂️

Reply:
One more time, Jane, thank you for your review. I'm appreciate that. Hope. you will help me by reviewing my NodeJS week 3 homework🤝🙏

Reply:
My pleasure!